File: ea/src/evolution_engine/selection/SelectionStrategy.java
package evolution_engine.selection;

import evolution_engine.evolution.Individual;
import evolution_engine.fitness.FitnessResult;

import java.util.List;

/**
 * Created by Itay Azaria
 * Date: 28/03/14
 * Time: 16:07
 */
public interface SelectionStrategy<I extends Individual> {
    public List<I> select(List<I> previousGeneration, FitnessResult fitness);
}

File: evolution/src/main/java/bgu/cs/evolution_engine/selection/SelectionStrategy.scala
package bgu.cs.evolution_engine.selection

import bgu.cs.evolution_engine.evolution.Individual
import bgu.cs.evolution_engine.fitness.FitnessResult
import scala.collection.immutable.List

/**
 * Created by Itay Azaria
 * Date: 28/03/14
 * Time: 16:07
 */
trait SelectionStrategy[I <: Individual] {
  def select(previousGeneration: List[I], fitness: FitnessResult[I]): List[I]
}